Flabbergasted-Flamingos/bangazon-commerce-app
# Cd search

# Description
Search functionality for products in the nav bar

## Number of Fixes
Type of number of fixes included in this PR.

## Related Ticket(s)
#1 #2 

## Problem to Solve
Search by product name and city

## Proposed Changes

## Expected Behavior
Describe the expected behavior.

## Steps to Test Solution
1. Run these in the terminal
```
git fetch --all
git checkout cd_search
rm -r db.sqlite3
rm -r website/migrations
python manage.py make migrations website
python manage.py migrate
python manage.py runserver
```
2. Open dbBrowser and run this sql
```
INSERT INTO auth_user VALUES (null, "f", 2018/8/18, 0, "ffff2", "ffff", "f@f.com", 1, 1, 2018/7/15, 'ffff');
INSERT INTO auth_user VALUES (null, "G", 2018/8/18, 0, "ggg2", "ggggg", "g@g.com",  0, 1, 2018/7/15,'ggg');
INSERT INTO website_paymentcategory VALUES (null,  null, 'Visa');
INSERT INTO website_paymentcategory VALUES (null, null, 'MasterCard');
INSERT INTO website_paymentcategory VALUES (null, null, 'AMEX');
INSERT INTO website_producttype VALUES (null, null, 'Home Goods');
INSERT INTO website_producttype VALUES (null, null, 'Pets');
INSERT INTO website_product VALUES (null, null, '', 'Poop-a-Rie', 12, "Make your poop not stink", '1',"Nashville", 1, 2);
INSERT INTO website_product VALUES (null, null, '', 'Febreeze', 12, "Make your house not stink", '1', "Nashville", 1, 2);
INSERT INTO website_product VALUES (null, null, '', 'Sandalwood Candle', 12, "Make your house like a delicious forest", '1', "Knoxville", 1, 1);
INSERT INTO website_product VALUES (null, null, '', 'Puppies', 12, "Australian Shepherd babies", '6', "Knoxville", 2, 2);
INSERT INTO website_product VALUES (null, null, '', 'Sugar Baby', 12, "Sweet Sugary Baby goodness", "Knoxville", '2', 2, 1);
INSERT INTO website_product VALUES (null, null, '' , 'Hamster', 12, "Furry cute rodents", '1', "Knoxville", 2, 1);
```

3. On the page, there should now be two input fields in the nav bar, one for product and one for city.
4. Run these 3 searches:
     product = Febreeze, city = blank
     product = blank, city = Knoxville
     product = Febreeze, city = Nashville
5. Each one should produce the results from the product table that match the search parameters
6. The name of each product should be a link that redirects to the product view page.

## Testing
NONE
[ ] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[ ] I certify that all existing tests pass

## Documentation
I HATE DOCUMENTATION
[ ] There is new documentation in this pull request that must be reviewed..
[ ] I added documentation for any new classes/methods

## Deploy Notes
Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.
